var/spack/repos/builtin/packages/libpng/package.py

```diff
@@ -29,8 +30,13 @@ class Libpng(CMakePackage):
     version("1.5.30", sha256="7d76275fad2ede4b7d87c5fd46e6f488d2a16b5a69dc968ffa840ab39ba756ed")
     version("1.2.57", sha256="0f4620e11fa283fedafb474427c8e96bf149511a1804bdc47350963ae5cf54d8")
 
-    depends_on("c", type="build")  # generated
-    depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+
+    depends_on("cmake@2.4.3:", type="build", when="@1.2.57")
+    depends_on("cmake@2.8.3:", type="build", when="@1.5.30")
+    depends_on("cmake@3.1:", type="build", when="@1.6.37:1.6.39")
+    depends_on("cmake@3.14:", type="build", when="@1.6.47:")
```

Comment:
```suggestion
    depends_on("cmake@3.14:", type="build", when="@1.6.47:")
    depends_on("cmake@3.1:", type="build", when="@1.6.37:")
    depends_on("cmake@2.8.3:", type="build", when="@1.5.30:")
    depends_on("cmake@2.4.3:", type="build", when="@1.2.57:")
```

These can be open ended constraints as Spack will figure out the endings when applying to a given version. In addition dependencies should be listed newest -> oldest, not oldest -> newest following with version ordering.